### ISSUE TITLE
Random `Variable not allowed in type expression`

### ISSUE THREAD

**Issue by minmax** [microsoft/pyright#9695]
`pyright 1.1.391`

I have a fairly large repository, with a lot of files (which makes it difficult to select a reproducible example).

In our CI there is strange behavior, periodically pyright printed error.

```
/app/name/types.py
/app/name/types.py:11:21 - error: Variable not allowed in type expression (reportInvalidTypeForm)
/app/name/types.py:13:19 - error: Variable not allowed in type expression (reportInvalidTypeForm)
```

and the file contents:

```python
from typing import Literal
from typing import TypeAlias


__all__ = ("FirstName", "SecondName")

FirstName: TypeAlias = Literal["FIRST_NAME"]

SecondName: TypeAlias = Literal["SECOND_NAME"]

FIRST_NAME: FirstName = "FIRST_NAME"

SECOND_NAME: SecondName = "SECOND_NAME"
```

I usually don't catch this error locally, but in CI it happens consistently once a month. Restarting helps, but I hope it's something "simple" and can be debugged without an example.

**Comment by erictraut** [comment]
You said that pyright crashes? If so, there should be a stack trace output. I'm guessing that this isn't a crash.

I suspect what you're saying is that pyright emits some diagnostics but does so intermittently (seemingly nondeterministically). Is that correct?

I would need repro steps to look into that. I'm not able to repro the problem with the code sample you've provided. There must be some additional details involved. Were you able to repro the issue with that code snippet? If so, is there anything else I need to do to make the issue occur?

**Comment by erictraut** [comment]
OK, if you're able to find a repro or additional clues, please feel free to add them to this report.

I'm going to close the issue at this time because there's nothing actionable without repro steps.
